src/retrieve-data-helpers/smart-launch.js

```diff
@@ -0,0 +1,48 @@
+import { signalSuccessFhirServerRetrieval } from '../actions/fhir-server-actions';
+import { signalSuccessSmartAuth, signalFailureSmartAuth } from '../actions/smart-auth-actions';
+import store from '../store/store';
+
+/**
+ * Handle SMART launch (Oauth2 response) accordingly if Sandbox is launched as a SMART on FHIR app.
+ * Store relevant data passed in.
+ * @returns {Promise} - Promise to resolve elsewhere
+ */
+function smartLaunchPromise() {
+  return new Promise((resolve, reject) => {
+    const onAuthFailure = () => {
+      console.log('Launched openly and unsecured');
+      store.dispatch(signalFailureSmartAuth());
+      return reject();
+    };
+
+    const onAuthSuccess = (smart) => {
+      console.log('Launched as a SMART app');
+      if (smart.tokenResponse) {
+        let fhirBaseUrl;
+        if (smart.server && smart.server.serviceUrl) {
+          fhirBaseUrl = smart.server.serviceUrl;
+          const conformanceQuery = smart.api.conformance({});
+          const deferredToPromise = Promise.resolve(conformanceQuery);
+          deferredToPromise.then((result) => {
+            store.dispatch(signalSuccessSmartAuth(smart));
```

Comment:
Could pass only the auth token since thats all we are storing in this action.

Reply:
Storing this object now because down the line, we may need to access other properties of `smart` like the user ID, the `api` field, etc. Also in the `smart.server.serviceUrl`, we get the base URL of the FHIR server which we need to store if this app is launched securely.

Reply:
👍